### PR TITLE
New template of probe command dispatch in hierarchy environment

### PR DESCRIPTION
**Pull request by hu-weihua**
What are this pull request do:
1. Extract command dispatch and reply collection part into a independent file, work as a class
2. Offer a probe sub_command template for customer, customer can use directly. this template  has finished the basic work what every sub_command have to do.

This template can run independently and output like below

    # xcatprobe hierarchy_template -n test[1-15]
    <mn>: I reveive node range is test9,test1,test2,test3,test4,test5                                                 [ OK ]
    <mn>: first checking point                                                                                        [ OK ]
    <mn>: second checking point                                                                                       [FAIL]
    <c910f02c04p04>: I reveive node range is test10,test11,test12,test13,test14,test15,test6,test7,test8              [ OK ]
    <c910f02c04p04>: first checking point                                                                             [ OK ]           
    <c910f02c04p04>: second checking point                                                                            [FAIL]
    ======================do summary=====================
    [mn]
        [ok]     : I reveive node range is test9,test1,test2,test3,test4,test5
        [ok]     : first checking point
        [failed] : second checking point
    [c910f02c04p04]
        [ok]     : I reveive node range is test10,test11,test12,test13,test14,test15,test6,test7,test8
        [ok]     : first checking point
        [failed] : second checking point

### CODE REVIEW

File: xCAT-probe/subcmds/code_template [hunk 1]
```diff
@@ -78,23 +55,19 @@ Options:
...
 # Recommand to use probe_utils->send_msg() to handle message you plan to print out
 # A simple example has been written in funciton.
 #------------------------------------
 sub do_main_job {
-    my $outputtarget = shift;
-    $outputtarget = "stdout" if (!$outputtarget);
     my $rst = 0;
 
-    probe_utils->send_msg($outputtarget, "o", "I reveive node range is $noderange");
+    probe_utils->send_msg("$output", "o", "I reveive node range is $noderange");
```
Comment (whowutwut) [inline xCAT-probe/subcmds/code_template]: I don't think we should use "I" in the message.  
Revive also should be "Received"? 

Suggest:
"Received node range: "

**Comment by whowutwut** [comment]
What is the first checking point and second checking point mean?

**Comment by hu-weihua** [comment]
@whowutwut ， thanks for your comments.

This is a template, also is a basic framework of how ``xcat probe`` to support hierarchy. 
For example, if a developer want to develop a new probe tool which will be used in hierarchy environment. he/she doesn't need to put effort to code for dispatching command to SNs and collect replay from SNs, this framework has done this part, he/she just needs to finish the main probe job of the new tool, just like this new tool just only work on  MN.

So ``first checking point and second checking point`` means the probe jobs which the new probe tool will do. Maybe this expression isn't easy to understand. How about using ``probe job 1 and probe job 2``. what do you think?

**Comment by xuweibj** [comment]
I agree to merge.

**Pull request merged**